scripts/util.lua

```diff
@@ -630,5 +630,24 @@ function module.replaceupvalue(func, name, replacement)
 	end
 end
 
+
+--- Format string
+--- With supports for "$" replace-able
+-- @str string to replace
+-- @param replacing fields
+-- @return replaced string
+local _StringFormat = string.format
+string.format = function(str, ...)
+	str2 = _StringFormat(str, ...)
+	if(str2 == str) then
+		for index, field in ipairs{...} do
```

Comment:
I can't think of a single instance where ipairs should be used over a index-based for loop in Insight's strings (or anywhere else, I am extremely hesitant when using ipairs). If I am passing in a nil value, I want it to error so it can be fixed, because I should never be passing in a nil value for string replacement.